### PR TITLE
Change pathing in test harness for windows

### PR DESCRIPTION
**Pull request by BenWilson2**
Signed-off-by: Ben Wilson <benjamin.wilson@databricks.com>

## What changes are proposed in this pull request?

Change the pathing to use os.path.join() instead of string interpolation for windows tests in test_file_store.py

## How is this patch tested?

unit tests

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [x] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Review by WeichenXu123 — APPROVED** [review]
LGTM

**Comment by BenWilson2** [comment]
Turns out the real root cause of this was the 'COM6' file name (awesome find @harupy !!!). Changing the test harness helper in `tests.helper_functions.random_str` to use `uuid.uuid4().hex[:size]` to preserve functionality of random variable length strings but preventing the possibility of protected windows namespace collisions as listed here: 
https://docs.microsoft.com/en-us/windows/win32/fileio/naming-a-file
Specifically these:
CON, PRN, AUX, NUL, COM1, COM2, COM3, COM4, COM5, COM6, COM7, COM8, COM9, LPT1, LPT2, LPT3, LPT4, LPT5, LPT6, LPT7, LPT8, and LPT9.
Which will make Windows very, very sad.

File: tests/helper_functions.py [hunk 1]
```diff
@@ -44,8 +43,12 @@ def random_int(lo=1, hi=1e10):
     return random.randint(lo, hi)
 
 
-def random_str(size=10, chars=string.ascii_uppercase + string.digits):
-    return "".join(random.choice(chars) for _ in range(size))
+def random_str(size=10):
+    """UUID4 generated strings have a high potential for collision at small sizes.
+    10 is lower bounds for this test harness utility."""
+    generated_string = uuid.uuid4().hex[:size]
+    assert len(generated_string) >= 10
```
Comment (harupy) [inline tests/helper_functions.py:50]:
```suggestion
    msg = (
        "UUID4 generated strings have a high potential for collision at small sizes. "
        "10 is lower bounds for this test harness utility."
    )
    assert size >= 10, msg
    generated_string = uuid.uuid4().hex[:size]
```

Can we add a message to `assert`?

**Review by harupy — APPROVED** [review]
LGTM once https://github.com/mlflow/mlflow/pull/5002#discussion_r742517964 is addressed!

File: tests/helper_functions.py (hunk 1 above)
Comment (BenWilson2) [reply tests/helper_functions.py]: changed!

**Pull request merged**